Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -1,147 +1,323 @@
+  simp only [finSepDegree, sepDegree_top]
+
+theorem sepDegree_bot' (K : Type w) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : Cardinal.lift.{v} (sepDegree F (⊥ : IntermediateField E K)) =
+      Cardinal.lift.{w} (sepDegree F E) := by
+  exact sepDegree_eq_of_equiv _ _ _ ((IntermediateField.botEquiv E K).restrictScalars F)
+
+@[simp]
+theorem sepDegree_bot'' (K : Type v) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : sepDegree F (⊥ : IntermediateField E K) = sepDegree F E := by
+  simpa only [Cardinal.lift_id] using sepDegree_bot' F E K
+
+@[simp]
+theorem finSepDegree_bot' (K : Type w) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : finSepDegree F (⊥ : IntermediateField E K) = finSepDegree F E := by
+  simpa only [Cardinal.toNat_lift] using congr_arg Cardinal.toNat (sepDegree_bot' F E K)
+
+@[simp]
+theorem sepDegree_top' (K : Type w) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : sepDegree F (⊤ : IntermediateField E K) = sepDegree F K := by
+  simpa only [Cardinal.lift_id] using sepDegree_eq_of_equiv _ _ _
+    ((IntermediateField.topEquiv (F := E) (E := K)).restrictScalars F)
+
+@[simp]
+theorem finSepDegree_top' (K : Type w) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : finSepDegree F (⊤ : IntermediateField E K) = finSepDegree F K := by
+  simp only [finSepDegree, sepDegree_top']
+
+/-- The standard isomorphism between `E →ₐ[F] K` and `E →ₐ[F] (normalClosure F E K)`. -/
+def algHomEquivNormalClosure (K : Type w) [Field K] [Algebra F K] :
+    (E →ₐ[F] K) ≃ (E →ₐ[F] (normalClosure F E K)) :=
+  ⟨fun f ↦ (IntermediateField.inclusion f.fieldRange_le_normalClosure) |>.comp
+    (AlgEquiv.ofInjectiveField f).toAlgHom, normalClosure F E K |>.val.comp,
+      fun _ ↦ rfl, fun _ ↦ rfl⟩
+
+/-- A random isomorphism between `Emb F E` and `E →ₐ[F] K` if `E = F(S)` such that every element
+`s` of `S` is integral (= algebraic) over `F` and whose minimal polynomial splits in `K`.
+Combined with `emb_nonempty`, it can be viewed as a stronger version of
+`IntermediateField.nonempty_algHom_of_adjoin_splits`. -/
+def embEquivOfAdjoinSplits (K : Type w) [Field K] [Algebra F K] {S : Set E}
+    (hS : adjoin F S = ⊤)
+    (hK : ∀ s ∈ S, IsIntegral F s ∧ Polynomial.Splits (algebraMap F K) (minpoly F s)) :
+    Emb F E ≃ (E →ₐ[F] K) :=
+  have halg := (topEquiv (F := F) (E := E)).isAlgebraic
+    (hS ▸ isAlgebraic_adjoin (S := S) <| fun x hx ↦ (hK x hx).1)
+  haveI := halg.isNormalClosure_normalClosure (L := AlgebraicClosure E) <| fun _ ↦
+    IsAlgClosed.splits_codomain _
+  haveI := halg.isNormalClosure_normalClosure (L := K) <| fun _ ↦
+    splits_of_mem_adjoin F (S := S) hK (hS ▸ mem_top)
+  algHomEquivNormalClosure F E (AlgebraicClosure E) |>.trans
+    (AlgEquiv.arrowCongr (@AlgEquiv.refl F E _ _ _) <| IsNormalClosure.equiv (K := E)) |>.trans
+      (algHomEquivNormalClosure F E K).symm
```

Comment:
Using [Algebra.IsAlgebraic.algHomEmbeddingOfSplits](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/NormalClosure.html#Algebra.IsAlgebraic.algHomEmbeddingOfSplits) and [Function.Embedding.antisymm](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Cardinal/SchroederBernstein.html#Function.Embedding.antisymm) will probably make a shorter proof. Having an explicit bijection is nice, but it's not like we provide good APIs for it to be useful ...